### DISCUSSION TITLE
Having Trouble Querying Using Datetime

### DISCUSSION THREAD

**Discussion by SureshKumarToverto** [capawesome-team/capacitor-firebase#591] [Q&A]
In Firebase, we have values in the format shown in the image below:
![image](https://github.com/capawesome-team/capacitor-firebase/assets/155732803/b7da5b71-623b-4db9-8453-4a815966f171)
 
 I am using the following function to retrieve dates:

public GetAllCheckInUser(filterData): any {
    var date =  moment(new Date()).subtract(1, 'h').toDate();
    console.log('date',date);    
    return from(FirebaseFirestore.getCollection({
      reference: 'LoggedinUser',
      compositeFilter: {
        type: 'and',
        queryConstraints: [         
          {
            type: 'where',
            fieldPath: 'LoggedinAt',
            opStr: '>',
            value: date,
          },
        ],
      },
    }))
      .pipe(
        map((actions) => actions.snapshots.map(a => {    
          const data = a.data as any;
          const id = a.id;
          return { id, ...data };
        })),
        map((data: any) => {
          return _.filter(data, (item) => item.AccountNumber !== null);
        })
      );
}

However, it's not working correctly. We used Firebase Firestore collection query referred from 
 https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/firestore
 
Thanks in advance

**Comment by robingenz** [comment]
This should be a duplicate of #474.
